apps/full-site-editing/full-site-editing-plugin/blocks/page-content/index.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { find, get, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { registerBlockType } from '@wordpress/blocks';
+import { IconButton, Placeholder, SelectControl, Toolbar } from '@wordpress/components';
+import { compose, withState } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const edit = compose(
+	withSelect( select => ( {
+		pages: select( 'core' ).getEntityRecords( 'postType', 'page', {
+			per_page: -1,
+		} ),
+	} ) ),
+	withState( {
+		isEditing: false,
+	} )
+)( ( { attributes, isEditing, pages, setAttributes, setState } ) => {
+	const { align, selectedPageId } = attributes;
+
+	const selectOptions = [
+		{ label: '', value: undefined },
+		...map( pages, page => ( {
+			label: `[${ page.id }] ${ page.title.rendered }`,
+			value: page.id,
+		} ) ),
+	];
+
+	const toggleEditing = () => setState( { isEditing: ! isEditing } );
+
+	const onChange = pageId => {
+		if ( pageId ) {
+			setState( { isEditing: false } );
+		}
+		setAttributes( { selectedPageId: parseInt( pageId, 10 ) } );
+	};
+
+	const selectedPage = find( pages, { id: selectedPageId } );
+
+	const showToggleButton = ! isEditing || !! selectedPageId;
+	const showPlaceholder = isEditing || ! selectedPageId;
+	const showPreview = ! isEditing && !! selectedPageId;
+
+	return (
+		<Fragment>
+			{ showToggleButton && (
+				<BlockControls>
+					<Toolbar>
+						<IconButton
+							className={ classNames( 'components-icon-button components-toolbar__control', {
+								'is-active': isEditing,
+							} ) }
+							label={ __( 'Change Preview' ) }
+							onClick={ toggleEditing }
+							icon="edit"
+						/>
+					</Toolbar>
+				</BlockControls>
+			) }
+			<div
+				className={ classNames( 'wpcom-page-content-block', {
+					[ `align${ align }` ]: align,
+				} ) }
+			>
+				{ showPlaceholder && (
+					<Placeholder
+						icon="layout"
+						label={ __( 'Page Content' ) }
+						instructions={ __( 'Select a page to preview' ) }
+					>
+						<div className="wpcom-page-content-block__selector">
+							<SelectControl
+								onChange={ onChange }
+								options={ selectOptions }
+								value={ selectedPageId }
+							/>
+							{ !! selectedPageId && (
+								<a href={ `?post=${ selectedPageId }&action=edit` }>{ __( 'Edit Page' ) }</a>
+							) }
+						</div>
+					</Placeholder>
+				) }
+				{ showPreview && (
+					<div
+						className="wpcom-page-content-block__preview"
+						// eslint-disable-next-line react/no-danger
+						dangerouslySetInnerHTML={ {
+							__html: get( selectedPage, 'content.rendered' ),
+						} }
+					/>
```

Comment:
there's a [`RawHTML` component](https://github.com/WordPress/gutenberg/blob/master/packages/element/src/raw-html.js) we should be able to use here. you can see where I've [used it in my prototype](https://github.com/dmsnell/stupid-code-do-not-use/blob/a8910a9720bb506841410c85b83429a1e932d5b2/fse/fse.js#L59).